src/Marten/Internal/ProviderGraph.cs

```diff
@@ -35,70 +39,78 @@ public ProviderGraph(StoreOptions options)
             return stored.As<DocumentProvider<T>>();
         }
 
-        if (documentType == typeof(IEvent))
+        lock (_storageLock)
```

Comment:
Could you extract a method for anything happening within that lock? Just for code tidy-ness.